app/utils/press-release-snapshot.js

```diff
@@ -1,39 +1,68 @@
 import { tracked } from '@glimmer/tracking';
 
+function serializePublicationChannels(publicationChannels) {
+  let publicationChannelsStringArray = [];
+  let publicationChannelsString = '';
+
+  publicationChannelsStringArray = publicationChannels.map(channel => channel.uri);
+
+  publicationChannelsString = publicationChannelsStringArray
+                                .sort()
+                                .join('+');
+
+  return publicationChannelsString;
+}
+
 /**
  * Snapshot of a PressRelease (Pers bericht) record and related records to keep track of changes,
  * because ember-data lacks dirty tracking for relationships and attributes of type 'array'.
  *
+ * Tracks relationship publicationChannels, to be able to find any changes, and detect those in the isDirty check.
+ * 
  * Contains application-specific logic to track the dirty state of relationships.
  * If the model of a PressRelease changes in the future, this class will probably require an update as well.
 */
 export default class PressReleaseSnapshot {
   @tracked pressRelease;
+  @tracked publicationChannels = [];
 
   constructor(pressRelease) {
     this.pressRelease = pressRelease;
   }
 
+
   async commit() {
-    // commit changes when pressRelease gets relations => e.g. source-snapshop.js
+    this.publicationChannels = await this.pressRelease.publicationChannels.map(channel => channel);
   }
 
   /**
    * Compares the snapshotted state of the PressRelease record with the current state.
    * Returns true if there is a difference.
   */
   async isDirty() {
-    return this.pressRelease.hasDirtyAttributes;
+    return this.pressRelease.hasDirtyAttributes ||
+      serializePublicationChannels(this.publicationChannels) !== serializePublicationChannels(await this.pressRelease.publicationChannels);
   }
 
   /**
    * Rollback the PressRelease record to the snapshotted state.
   */
   async rollback() {
     this.pressRelease.rollbackAttributes();
+    this.pressRelease.publicationChannels = this.publicationChannels;
   }
 
   async save() {
+    let publicationChannels = await this.pressRelease.publicationChannels;
+    let publicationEvent = await this.pressRelease.publicationEvent;
+    if(publicationEvent) {
+      publicationEvent.publicationChannels = publicationChannels;
+    }
+
+    await Promise.all([
+      publicationChannels.save(),
```

Comment:
Deze `save` lijkt mij overbodig. Resulteert dit in een network request?

Aan de attributen van een publication-channel kan niets gewijzigd worden via de app. De relatie van persbericht/publication-event naar publication-channel wordt opgeslagen aan de kant van persbericht/publication-event (via de aangepast serializer) dus deze call lijkt mij overbodig.

Reply:
resulteert in een netwerk request ja, maar is zoals je zegt overbodig.
zelfde functionaliteit als dit request niet gedaan wordt, verloopt idd via serializer